### PR TITLE
Feat: Integrate Request in parallel system (#32)

### LINKED ISSUE #32: Integrate Request in parallel system
Integrate system which throttles the request proccessed per second.
Hereby all schedulers share a pool of max requests per second which they can not overload, except the `DeadlineScheduler`m which is always able to execute as much as it needs.

Tasks:
- Integrate Endpoint to set the max number of requests 
- Integrate the requests into the shared`Nexus`
- Integrate system into `BulkScheduler` and `DeadlineScheduler`

### LINKED ISSUE #35: Integrate Github Test and Lint Action
Integrate Github Test and lint pipeline to prevent merging error prone branches.

Tasks:
- Setup pipeline
- Test
- Lint

### PR DESCRIPTION
**Pull request by KonsumGandalf**
Integrate system which throttles the request proccessed per second.
Hereby all schedulers share a pool of max requests per second which they can not overload, except the `DeadlineScheduler`m which is always able to execute as much as it needs.

Tasks:
- Integrate Endpoint to set the max number of requests 
- Integrate the requests into the shared`Nexus`
- Integrate system into `BulkScheduler` and `DeadlineScheduler`

Additional Task:
- Add Github pipeline

### CODE REVIEW

File: pkg/nexus/bulk/scheduler/bulk-scheduler.go [hunk 1]
```diff
@@ -53,7 +53,7 @@ func (ds *BulkScheduler) executeSchedule() {
 		}
```
Comment (Persists) [inline pkg/nexus/bulk/scheduler/bulk-scheduler.go:55]: We should take care of the loop, so it not also sleeps, when there are items in the queue, but not enaugh to execute as a bulk

File: pkg/nexus/bulk/scheduler/bulk-scheduler.go [hunk 2]
```diff
@@ -53,7 +53,7 @@ func (ds *BulkScheduler) executeSchedule() {
...
 
 		log.Println("Checking for bulking")
-		if itemsToPop := ds.Queue.GetMostCommonEntryItems(); len(itemsToPop) >= ds.MinAmountOfBulkItems {
+		if itemsToPop := ds.Queue.GetMostCommonEntryItems(); len(itemsToPop) >= ds.MinAmountOfBulkItems && ds.BaseNexusScheduler.MaxParallelRequests.Load() > 0 {
 			log.Println("items with name: " + itemsToPop[0].Name)
 			ds.Queue.RemoveAll(itemsToPop)
```
Comment (Persists) [inline pkg/nexus/bulk/scheduler/bulk-scheduler.go]: Remove doesnt trigger the tasks to be executed?

File: pkg/nexus/common/utils/helper-functions_test.go [hunk 3]
```diff
@@ -15,7 +15,7 @@ type HelperSuite struct {
 func (helperSuite *HelperSuite) TestGetEnvironmentHost() {
 	hostEnv := GetEnvironmentHost()
 
-	helperSuite.Equal("localhost", hostEnv)
+	helperSuite.Equal("host.docker.internal", hostEnv)
```
Comment (Persists) [inline pkg/nexus/common/utils/helper-functions_test.go]: please dont hardcode "host.docker.internal"

File: pkg/nexus/common/utils/helper-functions_test.go [hunk 4]
```diff
@@ -36,7 +36,7 @@ func (helperSuite *HelperSuite) TestTransformRequestToClientRequest() {
 	helperSuite.Equal("GET", transformedRequest.Method)
 	helperSuite.Equal("test", transformedRequest.Header.Get(headers.FunctionName))
 	helperSuite.Empty(transformedRequest.Header.Get(headers.ProcessDeadline))
-	helperSuite.Equal("http://localhost:8070/api", transformedRequest.URL.String())
+	helperSuite.Equal("http://host.docker.internal:8070/api", transformedRequest.URL.String())
```
Comment (Persists) [inline pkg/nexus/common/utils/helper-functions_test.go]: please dont hardcode "host.docker.internal"

File: pkg/nexus/nexus/nexus-router.go [hunk 5]
```diff
@@ -25,6 +26,7 @@ func (nexusRouter *NexusRouter) Initialize() {
 	nexusRouter.Router.Post("/scheduler/{schedulerName}/start", nexusRouter.StartScheduler)
 	nexusRouter.Router.Post("/scheduler/{schedulerName}/stop", nexusRouter.StopScheduler)
 	nexusRouter.Router.Get("/scheduler", nexusRouter.GetAllSchedulersWithStatus)
+	nexusRouter.Router.Put("/nexus-config", nexusRouter.ModifyNexusConfig)
```
Comment (Persists) [inline pkg/nexus/nexus/nexus-router.go]: now the route is /api/nexus/nexus-config

maybe use /api/nexus/config
-> 	nexusRouter.Router.Put("/config", nexusRouter.ModifyNexusConfig)

File: pkg/nexus/common/utils/helper-functions_test.go (hunk 3 above)
Comment (KonsumGandalf) [reply pkg/nexus/common/utils/helper-functions_test.go]: This is a test not a function

File: pkg/nexus/common/utils/helper-functions_test.go (hunk 4 above)
Comment (KonsumGandalf) [reply pkg/nexus/common/utils/helper-functions_test.go]: Again this is a test

File: pkg/nexus/nexus/nexus-router.go (hunk 5 above)
Comment (KonsumGandalf) [reply pkg/nexus/nexus/nexus-router.go]: Guter Hinweis danke

File: pkg/nexus/common/utils/helper-functions_test.go (hunk 3 above)
Comment (KonsumGandalf) [reply pkg/nexus/common/utils/helper-functions_test.go]: And on linux it works too

File: pkg/nexus/bulk/scheduler/bulk-scheduler.go (hunk 2 above)
Comment (KonsumGandalf) [reply pkg/nexus/bulk/scheduler/bulk-scheduler.go]: Ur right, came from the issue in rebasing previously. Thats why previous calling was lost. I wrote the functionality new.

File: pkg/nexus/bulk/scheduler/bulk-scheduler.go (hunk 1 above)
Comment (KonsumGandalf) [reply pkg/nexus/bulk/scheduler/bulk-scheduler.go:55]: I refactored the logic, does the new code looks better now?

File: pkg/nexus/common/utils/helper-functions_test.go (hunk 4 above)
Comment (KonsumGandalf) [reply pkg/nexus/common/utils/helper-functions_test.go]: Du hattest Recht sry.

**Comment by Persists** [comment]
lgtm

**Comment by Persists** [comment]
lgtm

**Pull request merged**